apiserver/facades/client/bundle/bundle.go

```diff
@@ -103,6 +103,62 @@ func (b *BundleAPI) checkCanRead() error {
 	return nil
 }
 
+// GetChanges returns the list of changes required to deploy the given bundle
+// data. The changes are sorted by requirements, so that they can be applied in
+// order.
+// V1 GetChanges did not support device.
+func (b *APIv1) GetChanges(args params.BundleChangesParams) (params.BundleChangesResults, error) {
+	var results params.BundleChangesResults
+	data, err := charm.ReadBundleData(strings.NewReader(args.BundleDataYAML))
+	if err != nil {
+		return results, errors.Annotate(err, "cannot read bundle YAML")
+	}
+	verifyConstraints := func(s string) error {
+		_, err := constraints.Parse(s)
+		return err
+	}
+	verifyStorage := func(s string) error {
+		_, err := storage.ParseConstraints(s)
+		return err
+	}
+	if err := data.Verify(verifyConstraints, verifyStorage, nil); err != nil {
+		if err, ok := err.(*charm.VerificationError); ok {
```

Comment:
I hope that different scopes will take effect here. However, from readability perspective, it seems to be difficult to differentiate between which 'err'...

Also could we please add tests that exercise this code path?